tools/booster/Proxy.hs

```diff
@@ -216,74 +245,93 @@ respondEither mbStatsVar booster kore req = case req of
-                -- We cannot call the booster internaliser without access to the server state
-                -- Again this should not fail.
-                let request =
-                        (emptyExecuteRequest simplified.state)
-                            { _module = mbModule
-                            , maxDepth = Just $ Depth 0
-                            }
-                Log.logInfoNS "proxy" "Making 0-step execute request to convert back to a term/constraints form"
-                result <- booster $ Execute request
-                case result of
-                    Right (Execute ExecuteResult{state = finalState}) -> pure finalState
-                    _other -> pure s
-            _other -> do
-                -- if we hit an error here, return the original
-                Log.logWarnNS "proxy" "Unexpected failure when calling Kore simplifier, returning original term"
-                pure s
-      where
-        toSimplifyRequest :: ExecuteState -> SimplifyRequest
-        toSimplifyRequest state =
-            SimplifyRequest
-                { state = execStateToKoreJson state
-                , _module = mbModule
-                , logSuccessfulSimplifications = Nothing
-                , logFailedSimplifications = Nothing
-                }
+    simplifyExecuteState ::
+        (Maybe Bool, Maybe Bool, Maybe Bool, Maybe Bool) ->
+        Maybe Text ->
+        ExecuteState ->
+        m (ExecuteState, Maybe [RPCLog.LogEntry])
+    simplifyExecuteState
+        ( logSuccessfulSimplifications
+            , logFailedSimplifications
+            , logSuccessfulRewrites
+            , logFailedRewrites
+            )
+        mbModule
+        s = do
+            Log.logInfoNS "proxy" "Simplifying execution state"
+            simplResult <-
+                handleSimplify $ toSimplifyRequest s
+            case simplResult of
+                -- This request should not fail, as the only possible
+                -- failure mode would be malformed or invalid kore
+                Right (Simplify simplified) -> do
+                    -- to convert back to a term/constraints form,
+                    -- we run a trivial execute request (in booster)
+                    -- We cannot call the booster internaliser without access to the server state
+                    -- Again this should not fail.
+                    let request =
+                            (emptyExecuteRequest simplified.state)
+                                { _module = mbModule
+                                , maxDepth = Just $ Depth 0
+                                }
+                    Log.logInfoNS "proxy" "Making 0-step execute request to convert back to a term/constraints form"
+                    result <- booster $ Execute request
+                    case result of
+                        Right (Execute ExecuteResult{state = finalState, logs = finalLogs}) ->
+                            pure (finalState, finalLogs)
```

Comment:
We would want to retain `simplified.logs` from the simplification response here. 
The `finalLogs` will always be empty because the 0-step request does not do any rewrites and all has been simplified before (or _should have been_...)

Reply:
Good catch! I will add simplification logs and ignore the 0-step execution logs

Reply:
On second though, I think we should retain the 0-step logs too, just in case...